src/test/regress/specs/isolation_drop_shards.spec

```diff
@@ -0,0 +1,60 @@
+setup
+{
+    CREATE TABLE append_table (test_id integer NOT NULL, data text);
+    SELECT create_distributed_table('append_table', 'test_id', 'append');
+
+    SELECT 1 FROM (
+      SELECT master_create_empty_shard('append_table') FROM generate_series(1,128)
+    ) a LIMIT 0;
+}
+
+teardown
+{
+    DROP TABLE append_table;
+}
+
+session "s1"
+
+step "s1-begin"
+{
+    BEGIN;
+}
+
+step "s1-apply-delete-command"
+{
+    SELECT master_apply_delete_command('DELETE FROM append_table');
+}
+
+step "s1-drop-all-shards"
+{
+    SELECT master_drop_all_shards('append_table', 'public', 'append_table');
+}
+
+step "s1-commit"
+{
+    COMMIT;
+}
+
+session "s2"
+
+step "s2-truncate"
+{
+    TRUNCATE append_table;
+}
+
+step "s2-apply-delete-command"
+{
+    SELECT master_apply_delete_command('DELETE FROM append_table');
+}
+
+step "s2-drop-all-shards"
+{
+    SELECT master_drop_all_shards('append_table', 'public', 'append_table');
+}
+
```

Comment:
OK. We can do it for `master_drop_all_shards`, though not for `master_apply_delete_command`, since SELECTs are allowed to proceed and then block on the worker, which means the isolation test doesn't proceed.